packages/cli/generation/ir-generator/src/utils/isGeneric.ts

```diff
@@ -0,0 +1,9 @@
+export function isGeneric(name: string): boolean {
+    const genericMatch = name.match(/([\w.]+)<([\w,\s]+)>/);
+
+    return (
+        genericMatch?.[0] != null &&
+        genericMatch[1] != null &&
+        !new Set(["optional", "set", "list", "map", "literal"]).has(genericMatch[1].trim())
```

Comment:
same here, can we be sharing this with existing logic?

Reply:
Consolidated a bit, where now `isGeneric` now leverages logic in `getGenericDetails`. Ideally there is a place where these utility functions could be shared between CLI packages. Taking a look for that